### PR TITLE
[Core]Append padding logic for Attention

### PR DESCRIPTION
**Pull request by Angazenn**
### What this PR does / why we need it?

This PR aims to add padding logic to seq_lens、block_tables when running in full decode scenario. Before this PR, the number of input tokens with padding might exceeds corresponding seq_lens. For example, when running in full decode scenario:

```
input_ids : [1, 3, 0, 0]
seq_lens: [2, 1]
query_start_loc: [0, 1, 2]
```
Here, `input_ids` is padded by 2 tokens while `seq_lens`/`query_start_loc` are not. The mismatch between `input_ids` and `seq_lens`/`query_start_loc` might cause some potential bugs. This PR would change it into :

```
input_ids : [1, 3, 0, 0]
seq_lens: [2, 1, 1, 1]
query_start_loc: [0, 1, 2, 3, 4]
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

- vLLM version: v0.11.0rc3
- vLLM main: https://github.com/vllm-project/vllm/commit/v0.11.0

### CODE REVIEW

File: vllm_ascend/attention/attention_v1.py [hunk 1]
```diff
@@ -207,6 +207,28 @@ def build(
...
+                device=block_table.device)
+            block_table = torch.cat([block_table, block_table_padding], dim=0)
+            query_start_loc_cpu = torch.cat([
+                query_start_loc_cpu,
+                torch.arange(query_start_loc_cpu[-1] + 1,
+                             query_start_loc_cpu[-1] + padded_num_tokens,
+                             dtype=query_start_loc_cpu.dtype,
+                             device=query_start_loc_cpu.device)
+            ])
+
```
Comment (yiz-liu) [inline vllm_ascend/attention/attention_v1.py:241]: Why don't we handle this in `prepare_inputs` — make everything in `common_attn_metadata` final so it doesn't need further modification? I'm worried this version will add maintenance overhead, since there are other attention implementations.

Comment (Angazenn) [reply vllm_ascend/attention/attention_v1.py:241]: I move this to `prepare_inputs` function. But I'm not sure whether these padding can generalize to different attention backend. More test is needed.

Comment (Angazenn) [reply vllm_ascend/attention/attention_v1.py:241]: It seems that moving these padding to `prepare_inputs` is in conflicts with mtp and there might be other potential bugs with different attention backends. So I decide to move it back to `build`.

File: vllm_ascend/worker/model_runner_v1.py [hunk 2]
```diff
@@ -1583,7 +1583,10 @@ def _build_attn_state(self, num_reqs, num_scheduled_tokens,
         return attn_state
 
     def _update_graph_pad_size(self, with_prefill, graph_pad_size):
-        self.graph_pad_size = -1
+        if not with_prefill:
+            self.graph_pad_size = graph_pad_size
+        else:
+            self.graph_pad_size = -1
```
Comment (yiz-liu) [inline vllm_ascend/worker/model_runner_v1.py]: Please use attn_state instead of modifying this method.

**Review by yiz-liu — CHANGES_REQUESTED** [review]

File: vllm_ascend/worker/model_runner_v1.py (hunk 2 above)
Comment (Angazenn) [reply vllm_ascend/worker/model_runner_v1.py]: done

**Review by yiz-liu — APPROVED** [review]

File: vllm_ascend/attention/attention_v1.py [hunk 3]
```diff
@@ -216,6 +216,29 @@ def build(
         query_start_loc_cpu = common_attn_metadata.query_start_loc_cpu[:
                                                                        num_reqs
                                                                        + 1]
+
+        if attn_state == AscendAttentionState.DecodeOnly and \
```
Comment (momo609) [inline vllm_ascend/attention/attention_v1.py:220]: why this only affect decodeonly state.
